### ISSUE TITLE
suspiciously_else_formatting triggered for right code

### ISSUE THREAD

**Issue by jdortiz** [rust-lang/rust-clippy#7024]
Lint name:
suspiciously_else_formatting

I tried this code:

```rust
#[post("/login", format = "application/json", data = "<user>")]
pub fn login(user: Json<User>) -> Result<Json<Token>, Status> {
    //! Some comment
    //! Some other comment
    let user = user.into_inner();
    if let Some(token) = security::login(user) {
        Ok(Json(Token { token }))
    } else {
        Err(Status::Unauthorized)
    }
}
```

I expected to see this happen: No warning emitted

Instead, this happened: This warning emitted
```
warning: this is an `else {..}` but the formatting might hide it
  --> worker-api/src/api/auth.rs:9:1
   |
9  | / pub fn login(user: Json<User>) -> Result<Json<Token>, Status> {
10 | |     //! Some comment
11 | |     //! Some other comment
12 | |     let user = user.into_inner();
...  |
17 | |     }
18 | | }
   | |_^
   |
   = note: `#[warn(clippy::suspicious_else_formatting)]` on by default
   = note: to remove this lint, remove the `else` or remove the new line between `else` and `{..}`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_else_formatting
```
### Meta

- `cargo clippy -V`: clippy 0.1.51 (2fd73fab 2021-03-23)
- `rustc -Vv`:
  ```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-apple-darwin
release: 1.51.0
LLVM version: 11.0.1
  ```

**Comment by jdortiz** [comment]
If I remove one of the two comments the warning goes away.

**Comment by extrawurst** [comment]
sounds related to #6249

**Comment by Jarcho** [comment]
This should have been fixed by #7707

**Comment by y21** [comment]
Closing as this no longer reproduces (presumably fixed by the PR that added the proc macro check like the other comments have said)
